cores/esp32/esp32-hal-uart.c

```diff
@@ -277,26 +278,24 @@ uint32_t uartAvailableForWrite(uart_t* uart)
     return 0x7f - uart->dev->status.txfifo_cnt;
 }
 
-uint8_t uartRead(uart_t* uart)
+uint8_t uartRead(uart_t* uart, uint8_t *data, size_t to)
 {
     if(uart == NULL || uart->queue == NULL) {
         return 0;
     }
-    uint8_t c;
-    if(xQueueReceive(uart->queue, &c, 0)) {
-        return c;
+    if(xQueueReceive(uart->queue, data, to)) {
+        return 1;
     }
     return 0;
 }
 
-uint8_t uartPeek(uart_t* uart)
+uint8_t uartPeek(uart_t* uart, uint8_t *data, size_t to)
 {
     if(uart == NULL || uart->queue == NULL) {
         return 0;
     }
-    uint8_t c;
-    if(xQueuePeek(uart->queue, &c, 0)) {
-        return c;
+    if(xQueuePeek(uart->queue, data, to)) {
+        return 1;
```

Comment:
same for this method, the uartPeek() method returns the next character available to read

Reply:
@atanisoft same here